src/Integration/LocalServices/ErrorListInfoBarController.cs

```diff
@@ -462,6 +491,190 @@ private ProjectViewModel FindProject(Uri serverUri, string projectKey)
+                    updateAction();
+                    return;
+                }
+
+                VsShellUtils.WriteToGeneralOutputPane(this.host, Strings.SonarLintProfileCheck);
+
+                CancellationToken token = this.TokenSource.Token;
+                this.BackgroundTask = System.Threading.Tasks.Task.Run(() =>
+                {
+                    if (this.ProcessQualityProfileChanges(binding, languages, token))
+                    {
+                        this.host.UIDispatcher.BeginInvoke(new Action(()=>
+                        {
+                            // We mustn't update if was cancelled
+                            if(!token.IsCancellationRequested)
+                            {
+                                updateAction();
+                            }
+                        }));
+                    }
+                }, token);
+            }
+
+            private bool ProcessQualityProfileChanges(BoundSonarQubeProject binding, IEnumerable<LanguageGroup> projectLanguageGroups, CancellationToken token)
+            {
+                Debug.Assert(binding != null);
+
+                ConnectionInformation connection = binding.CreateConnectionInformation();
+                Dictionary<LanguageGroup, QualityProfile> newProfiles;
+                if (!this.TryGetNewProfiles(binding, projectLanguageGroups, token, connection, out newProfiles))
+                {
+                    VsShellUtils.WriteToGeneralOutputPane(this.host, Strings.SonarLintProfileCheckFailed);
+                    return false; // Error, can't proceed
+                }
+
+                if (!newProfiles.Keys.All(binding.Profiles.ContainsKey))
+                {
+                    VsShellUtils.WriteToGeneralOutputPane(this.host, Strings.SonarLintProfileCheckSolutionRequiresMoreProfiles);
+                    return true; // Missing profile, refresh
+                }
+
+                foreach (var keyValue in binding.Profiles)
+                {
+                    LanguageGroup group = keyValue.Key;
+                    ApplicableQualityProfile oldProfileInfo = keyValue.Value;
+
+                    if (!newProfiles.ContainsKey(group))
+                    {
+                        // Not a relevant profile, we should just ignore it.
+                        continue;
+                    }
+
+                    QualityProfile newProfileInfo = newProfiles[group];
+                    if (IsNewProfileRequriesAnUpdate(newProfileInfo, oldProfileInfo))
+                    {
+                        return true;
+                    }
+                }
+
+                VsShellUtils.WriteToGeneralOutputPane(this.host, Strings.SonarLintProfileCheckQualityProfileIsUpToDate);
+                return false; // Up-to-date
+            }
+
+            private bool IsNewProfileRequriesAnUpdate(QualityProfile newProfileInfo, ApplicableQualityProfile oldProfileInfo)
```

Comment:
Typo - "Requries".
The name is rather cumbersome - does it just mean _HasProfileChanged_?